docs/upgrade.html

```diff
@@ -60,6 +60,7 @@
     <li> The new Java consumer is no longer in beta and we recommend it for all new development. The old Scala consumers are still supported, but they will be deprecated in the next release
          and will be removed in a future major release. </li>
     <li> The BrokerState "RunningAsController" (value 4) has been removed. Due to a bug, a broker would only be in this state briefly before transitioning out of it and hence the impact of the removal should be minimal. The recommended way to detect if a given broker is the controller is via the kafka.controller:type=KafkaController,name=ActiveControllerCount metric. </li>
+    <li> Cluster Id has been added. It will be generated automatically on upgrading and restarting the brokers. It is available as kafka.server:type=KafkaServer,name=ClusterId metric and is also part of the Metadata response. Any class can get access to cluster id by implementing the ClusterResourceListener interface. </li>
```

Comment:
Can you add a "New Protocol Versions" section" after "Notable changes" and include information about the `MetadataResponse` change and move `ListOffsetRequest` a few lines above to that section too?

Then this line can become:

"Kafka clusters can now be uniquely identified by a cluster id. It will be automatically generated when a broker is upgraded to 0.10.1.0. The cluster id is available via the <code>kafka.server:type=KafkaServer,name=ClusterId</code> metric and it is part of the Metadata response. Serializers, client interceptors and metric reporters can receive the cluster id by implementing the <code>ClusterResourceListener</code> interface."